sql/src/main/java/io/crate/protocols/postgres/BatchPortal.java

```diff
@@ -126,7 +126,11 @@ public void execute(ResultReceiver resultReceiver, int maxRows) {
     public void sync(Planner planner, StatsTables statsTables, CompletionListener listener) {
         BatchCompletionListener batchCompletionListener = new BatchCompletionListener(analysis.size(), listener);
         for (int i = 0; i < analysis.size(); i++) {
-            UUID jobId = UUID.randomUUID();
+            UUID jobId = sessionData.getJobId();
+            if (i > 0) {
+                jobId = UUID.randomUUID();
+                statsTables.jobStarted(jobId, queries.get(i));
```

Comment:
Currently it's the portal.bind() that detects the batch operation so we can have a List of jobIds in the BatchPortal and add a new jobId in every bind and in the sync call we use this list instead of creating the jobIds there.
Any better suggestion?